src/test/java/io/github/sashirestela/slimvalidator/validators/ExtensionValidatorTest.java

```diff
@@ -59,6 +59,90 @@ void shouldThrownExceptionWhenNonExpectedObjectIsPassed() {
         assertEquals(expectedMessage, actualMessage);
     }
 
+    @Test
+    void shouldThrownExceptionForHiddenFilesInFile() {
+        var validator = new ExtensionValidator();
+        var annotation = Sample.extension(new String[] { "mp3", "wav" });
+        validator.initialize(annotation);
+        File file = new File(".hiddenfile");
+        var exception = assertThrows(ValidationException.class, () -> validator.isValid(file));
+        var actualMessage = exception.getMessage();
+        var expectedMessage = "No valid file extension found.";
+        assertEquals(expectedMessage, actualMessage);
+    }
+
+    @Test
+    void shouldThrownExceptionForHiddenFilesInPath() {
+        var validator = new ExtensionValidator();
+        var annotation = Sample.extension(new String[] { "mp3", "wav" });
+        validator.initialize(annotation);
+        File file = new File(".hiddenfile");
+        var exception = assertThrows(ValidationException.class, () -> validator.isValid(file.toPath()));
+        var actualMessage = exception.getMessage();
+        var expectedMessage = "No valid file extension found.";
+        assertEquals(expectedMessage, actualMessage);
+    }
+
+    @Test
+    void shouldThrownExceptionForEmptyFilenameInFile() {
+        var validator = new ExtensionValidator();
+        var annotation = Sample.extension(new String[] { "mp3", "wav" });
+        validator.initialize(annotation);
+        File file = new File("");
+        var exception = assertThrows(ValidationException.class, () -> validator.isValid(file));
+        var actualMessage = exception.getMessage();
+        var expectedMessage = "File name is null or empty.";
+        assertEquals(expectedMessage, actualMessage);
+    }
+
+    @Test
+    void shouldThrownExceptionForEmptyFilenameInPath() {
+        var validator = new ExtensionValidator();
+        var annotation = Sample.extension(new String[] { "mp3", "wav" });
+        validator.initialize(annotation);
+        File file = new File("");
+        var exception = assertThrows(ValidationException.class, () -> validator.isValid(file.toPath()));
```

Comment:
Only one method invocation is expected when testing runtime exceptions. Check Sonar findings:

https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=17&id=sashirestela_slimvalidator&open=AZGVmwUKmvjcF6WpIfCG